### PR TITLE
fix: Fixing UNC path GetFullPath when Current Directory is a UNC path

### PR DESCRIPTION
**Pull request by michal-ciechan**
Currently, if you switch current directory in the `MockFileSystem` to a UNC path, and try to enumerate files, it blows up.

Attempting to fix it, although it feels like a hack.

### CODE REVIEW

File: src/System.IO.Abstractions.TestingHelpers/MockPath.cs [hunk 1]
```diff
@@ -115,13 +115,16 @@ public override string GetFullPath(string path)
             {
                 fullPath = DirectorySeparatorChar + fullPath;
             }
-            else if (isUnixRooted)
+            else if (isUnixRooted || isUnc)
```
Comment (michal-ciechan) [inline src/System.IO.Abstractions.TestingHelpers/MockPath.cs:118]: feels like we should only set isUnixRooted if "current directory is not a UNC path" which would resolve this issue as well. Although re-using the same slashes feels safest

Comment (fgreinacher) [reply src/System.IO.Abstractions.TestingHelpers/MockPath.cs:118]: I agree, that sounds more correct.  I think `isUnixRooted` should only be `true` when the path starts with a forward slash.

Comment (michal-ciechan) [reply src/System.IO.Abstractions.TestingHelpers/MockPath.cs]: Done, much cleaner

**Review by fgreinacher — APPROVED** [review]
Thanks a lot 👍

**Pull request merged**